scipy/stats/_unuran/unuran_wrapper.pyx.templ

```diff
@@ -918,6 +924,428 @@ cdef class TransformedDensityRejection(Method):
+           https://statmath.wu.ac.at/software/unuran/doc/unuran.html#PINV
+
+    Examples
+    --------
+    >>> from scipy.stats import NumericalInversePolynomial
+    >>> from scipy.stats import norm
+
+    To create a generator to sample from the standard normal distribution, do:
+
+    >>> class StandardNormal:
+    ...    def pdf(self, x):
+    ...        return np.exp(-0.5 * x*x)
+    ... 
+    >>> dist = StandardNormal()
+    >>> urng = np.random.default_rng()
+    >>> rng = NumericalInversePolynomial(dist, random_state=urng)
+
+    Once a generator is created, samples can be drawn from the distribution by calling
+    the `rvs` method:
+
+    >>> rng.rvs()
+    -1.5244996276336318
+
+    As this is an invertion method, the approximated PPF can be evaluated by calling
+    the `ppf` method:
+
+    >>> rng.ppf(0.975)
+    1.9599639857012559
+    >>> norm.ppf(0.975)
+    1.959963984540054
+
+    We can see how well the generator approximates the standard normal distribution by
+    plotting the error between approximated PPF and exact PPF of the distribution:
+
+    >>> import matplotlib.pyplot as plt
+    >>> u = np.linspace(0.01, 0.99, 1000)
+    >>> approxppf = rng.ppf(u)
+    >>> exactppf = norm.ppf(u)
+    >>> error = np.abs(exactppf - approxppf)
+    >>> plt.plot(u, error)
+    >>> plt.xlabel('u')
+    >>> plt.ylabel('error')
+    >>> plt.title('Error between exact and approximated PPF')
+    >>> plt.show()
+
+    During setup, the CDF of the distribution is computed numerically but it is
+    discarded once the setup is complete. To keep the approximated CDF, pass
+    ``keep_cdf=True`` during initialization:
+
+    >>> rng = NumericalInversePolynomial(dist, keep_cdf=True, random_state=urng)
+
+    Now, the CDF can be evaluated by calling the `cdf` method:
+
+    >>> rng.cdf(1.959963984540054)
+    0.9750000000042454
+    >>> norm.cdf(1.959963984540054)
+    0.975
+
+    It is also possible to extimate the u-error of the approximated PPF if the exact
+    CDF is available during setup. To do so, pass a `dist` object with exact CDF of
+    the distribution during initialization:
+
+    >>> from scipy.special import ndtr
+    >>> class StandardNormal:
+    ...    def pdf(self, x):
+    ...        return np.exp(-0.5 * x*x)
+    ...    def cdf(self, x):
+    ...        return ndtr(x)
+    ... 
+    >>> dist = StandardNormal()
+    >>> urng = np.random.default_rng()
+    >>> rng = NumericalInversePolynomial(dist, random_state=urng)
+
+    Now, the u-error can be estimated by calling the `u_error` method. It runs a
+    Monte-Carlo simulation to estimate the u-error. By default, 100000 samples are
+    used. To change this, you can pass the number of samples as an argument:
+
+    >>> rng.u_error(sample_size=1000000)  # uses one million samples
+    UError(max_error=8.785994154436594e-11, max_absolute_error=2.930890027826552e-11)
+
+    This returns a namedtuple which contains the maximum u-error and the maximum
```

Comment:
MAE = mean absolute error. should we say `mae_error` => this is the mean of the absolute errors computed over the given sample

Reply:
> should we say `mae_error`

As we have been quite explicit in the design of this API, it would be better to cal it `mean_absolute_error` (that's was I was thinking in the first place, `max_absolute_error` is just a silly mistake)

Reply:
Agree, let's go with "mean_absolute_error"

Reply:
Done.